scipy/stats/morestats.py

```diff
@@ -2202,8 +2236,8 @@ def ansari(x, y):
         else:  # N even
             varAB = m * n * (16*fac - N*(N+2)**2) / (16.0 * N * (N-1))
 
-    z = (AB - mnAB) / sqrt(varAB)
-    pval = distributions.norm.sf(abs(z)) * 2.0
+    z = (mnAB - AB) / sqrt(varAB)
```

Comment:
This really threw me for a loop. The test statistic AB is actually defined such that it increases as scale(y)/scale(x) increases. This is opposite to the way we define the ratio of scales. There should be a comment about that in the code.

Reply:
> This really threw me for a loop. The test statistic AB is actually defined such that it increases as scale(y)/scale(x) increases. This is opposite to the way we define the ratio of scales. There should be a comment about that in the code. It makes sense given the definition of the test statistic, but if the reader is not familiar with the definition, then it looks surprising.

I also read this in the original paper (`[1]_`). I will write a comment about this and mention the reference.

Reply:
I have now added a comment explaining this.